docs/infrastructure/github-secrets.md

```diff
@@ -0,0 +1,37 @@
+## GitHub Secrets
+
+GitHub secrets are intended for use by GitHub Actions workflows. If you want a workflow to interact with something that requires authentication, you will likely need to create a secret which will allow authentication to happen securely without making any private information available to anybody having access to the repositories.
+
+There are three levels that GitHub secrets can exist at: organization, repo, and environment.
+
+An organization secret belongs to the `eclipse-pass` organization and is available to workflows that are part of any repository that exists under the `eclipse-pass` organization. Secrets that will be used across many repositories should be placed here to prevent duplication.
+
+A repository secret exists within that repository and can only be used by workflows belonging to that repository.
+
+An environment secret exists within a specific environment within a repository. It can only be used by workflows belonging to that repository and the workflow must specify the environment that it's using.
+
+### Creating Secrets
+
+GitHub offers two mechanisms for creating secrets: UI and API.
+
+The GitHub UI requires that a person creating a secret have admin privileges for that entity (either org or repo). Since members of the PASS project don't have admin privileges at any level, we need to open a ticket with Eclipse to add secrets through the UI.
+
+Oddly, the GitHub API allows the creation of repository secrets with `repo` privileges. All members of the PASS project should have sufficient privileges to do so.
+
+To create a repository or environment secret, use the script found at `main/tools/github_secrets.py` like so:
```

Comment:
You can reference this as a link instead of a "code"

[/tools/github_secrets.py](/tools/github_secrets.py)

Reply:
You can test it "once" it's in your branch to ensure it "maintains" the tree information.